### DISCUSSION TITLE
[Unstable] Wreck hulls 2

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#13719] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Following https://github.com/FakeFishGames/Barotrauma/issues/13524, a lot of wreck hulls were removed from wrecks.

Unfortunately I think my former post was misunderstood, so I'll elaborate here:

**DRAW BEHIND SUBS LIGHTS WILL SHINE THROUGH UNHULLED SECTIONS OF WRECKS.**
Any sub turret is set as draw behind for performance reasons. When a diver is inside the wreck, they see this:

## BAD
The wreck is lit up despite being surrounded by walls and background walls
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/627efb07-7962-4e02-90c2-82bfe04fb414)
You can see a room surrounded by walls being lit up at the top, and a hard cutoff at the bottom
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/1a8d2f4d-237b-4819-904d-cf8ef5342279)

## GOOD:
1. Large open breach
2. Breached from the top
3. The decorations have limited background walls

The light shining "inside" is plausible
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/e5f7ec36-a1e7-443a-9a2b-ff7d65e6f2e8)
Typhon2 was a good change
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/1498b1a2-245c-4be9-8f42-10b06f4037ab)

 ## FIXES
 

Hull whenever possible 
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/a5d576c1-1bdc-4e9a-81d3-4213908015d8)

Reduce the "contact" area to a small gap/entrance and avoid corners
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/fea0e19d-ae3b-4327-bce6-870f671e51ee)

Example that mixes both:

Before
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/608acb66-0c20-4eb0-8f40-0f6bd804d86a)

After
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/6fe8a2ea-6349-4cd8-84fd-6c6e73e85220)
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/92f780b5-626c-431c-aa85-c4cd59a02fe8)

### Version

v1.4.0.0 (unstable)

**Comment by PanmanS** [comment]
Nah, his hulls suck

**Comment by PanmanS** [comment]
If anything just hull camel gunnery


Basically try to have no hull only in areas with breaches untill first door/hatch

**Comment by PanmanS** [comment]
Dont let this goober take away muh glorious hulls

**Comment by Regalis11** [comment]
While I agree the way lights shine through "non-hulled" spaces looks off in these cases, and that we might need to address that, I feel it's largely subjective what the "correct" way to set up the hulls is.
